Style improvements

aliblackwell opened the pull request:
- CSS vars with IE11 ponyfill
- consistent styles
- accessible loading animation

aliblackwell commented:
Hey @chris48s, this is good to go and I will continue to improve things off this base in my branches for accessibility and translations.

chris48s commented:
Cheers for this. Looks really good. A few bits of feedback, but its very subtle stuff:

For the error style, I'd like to avoid re-using the "hot pink" style we use for links/accents for the errors. Can we go with something closer to the DC theme styles (which uses #b10e1e ) here for the error box. This is the equivalent error on wheredoivote.co.uk :

![Screenshot at 2019-10-22 16-42-46](https://user-images.githubusercontent.com/6025893/67306336-b6561780-f4ee-11e9-9df4-6b0d8c6cf42c.png)

---

This is quite a subtle point, but I reckon we could do with a bit more vertical space between either the bottom of the polling station address text and the "show me directions" button, or possibly between the "show me directions" button and "back button" (depending if its staying in that position - see next point). In a case like this one, I don't think it matters very much:

![Screenshot at 2019-10-22 16-44-21](https://user-images.githubusercontent.com/6025893/67306368-c241d980-f4ee-11e9-8cef-cd2bf0de2c7e.png)

but in a case like this one where we don't have a well formatted address and there's some other content above, a bit more vertical spacing under the address might help that content to breathe a bit more.

![Screenshot at 2019-10-22 16-43-23](https://user-images.githubusercontent.com/6025893/67306395-ccfc6e80-f4ee-11e9-87ce-66070864f32e.png)

Another approach we could take here that might focus that a bit more would be to group the
"Your Polling Station" / [station address] / "show me directions" or
"We couldn't find your station" / [council contact details]
block into a bordered box too (with the same spacing as the notification)?
Not sure if that would look good or too busy..

Also in this case (where we're showing a polling station address and a notification), I reckon we want the polling station address text to be either the same font size or slightly larger than the notification text (at the moment, it looks like the notification text is slightly bigger). The notification is really important, but the address of the polling station is still the key content: its the one bit of info the user came here to find, so we don't want the notification to be prominent but not _completely_ distract focus.

I'm aware that I've chosen an example where we have no line breaks in the address to deliberately show the absolute worst-case scenario here.

---

What are your thoughts on styling for the back button? Is that your final suggestion, or just a placeholder for "do something better"?

aliblackwell commented:
Very much a placeholder for "do something better"! I will have a play with some of the muted colours and see what works, and also look further at the wheredoivote.co.uk site for inspiration.

I did wonder about using the pink for an error message. I will revert to the red.

Excellent point about the spacing. I do anticipate further improvements as we work through the accessibility and translations cards. I will however address these points tomorrow morning and get this PR ship-shape before it's merged in.

aliblackwell commented:
@chris48s I am happy with this now as a base for future development. 

I have moved notifications and errors below the main content. I think this nicely addresses your point above about priority. I don't think boxing it in will work, particularly if we're going to have lists of candidates in future. What do you think?

I've also implemented a muted secondary button style. Perhaps the hover state is a touch too much?

Otherwise good to go!

chris48s commented:
Good progress. Moving the notifications down works, but I reckon in the case where its an _error_ (e.g: "that's not a valid postcode"), as opposed to a _notification_, that should stay as the top item of content (i.e: above the postcode search dialog) as that's the most important content in that case.

Like the new style for the back button, there are a few other problems though:

### Address picker

Can we make sure the button is always below the address picker, even if its really small. This looks odd:

![Screenshot at 2019-10-25 08-37-24](https://user-images.githubusercontent.com/6025893/67553338-d8c17e00-f704-11e9-8b73-a5d50e2c8c01.png)

If the picker contains really long addresses, the picker should not overflow the parent div:

![Screenshot at 2019-10-25 08-39-57](https://user-images.githubusercontent.com/6025893/67553347-e2e37c80-f704-11e9-8743-31219b7d0835.png)

### Notification

We've lost some spacing between the `i` symbol and heading. This looks a bit odd:

![Screenshot at 2019-10-25 08-39-06](https://user-images.githubusercontent.com/6025893/67553417-04dcff00-f705-11e9-9e3b-6e64ea02c511.png)

### Search Box

Bold design choice, but I think we need the outline back. To someone who has never seen it before, I don't think its clear where the text box is going to be (or that there is one).

![Screenshot at 2019-10-25 08-39-28](https://user-images.githubusercontent.com/6025893/67553453-18886580-f705-11e9-9e42-7491bbd5dd4a.png)

aliblackwell commented:
Eek good spots on all of that. I did some final tidying (searching and replacing variable names) and didn't check it afterwards. Box is back on the input, there's more space around the icon, the button is sitting pretty, and the address picker has a new max-width. I'll boot up my VM and check it all in IE11 as well.
